### ISSUE TITLE
Add option to process newest Action first

### ISSUE THREAD

**Issue by dannysellers** [coosto/ShortcutJS#22]
ShortcutJS indexes Actions by name, rather than KeyCombo, allowing multiple Actions to be bound to the same shortcut (i.e. using Escape to trigger `closeSearchBar`, and sometimes to `closeModal`). When processing action combos, the EventProcessor iterates over matching Actions in order, triggering combos on the first Action registered. Actions do support triggering multiple callbacks, but I propose adding an option to process multiple Actions bound to the same KeyCombo, in LIFO order.

For example, In a React component, Actions may be created, registered, and processed all within a single context for semantic reasons (i.e. all Search-related actions inside a Search component). Consider the following example:

```
class SearchBar extends React.PureComponent {
  constructor () {
    this.state = { hidden: true };
  }

  componentDidMount () {
    const closeSearchAction = new Action('closeSearch', KeyCombo.fromString('escape'));

    shortcutJS.addAction(closeSearchAction);
    shortcutJS.subscribe('closeSearch', this.setState({ hidden: true }); // showing controlled elsewhere
  }

  render () {
    let className = "search-bar";
    if (this.state.hidden) {
      className += " hidden";
    }

    return <div class={className}>
      <input type="search" />
    </div>;
  }
}

class Modal extends React.PureComponent {
  constructor () {
    this.state = { hidden: true };
  }

  componentDidMount () {
    const closeModalAction = new Action('closeModal', KeyCombo.fromString('escape'));

    shortcutJS.addAction(closeModalAction);
    shortcutJS.subscribe('closeModal', this.setState({ hidden: true }); // showing controlled elsewhere
  }

  render () {
    let className = "modal";
    if (this.state.hidden) {
      className += " hidden";
    }

    return <div class={className}>
      {this.props.modalContent}
    </div>;
  }
}
```
Tl;dr: Two disparate components have behaviors linked to the same KeyCombo, but because the `SearchBar` is instantiated before the `Modal`, the modal's Action never gets processed on pressing `escape`.

Given that these are disparate components, and that Actions are named semantically (according to the docs), it doesn't make sense to do `shortcutJS.actions.get('closeSearch').callbacks.add(() => this.setState({ hidden: true }));` from within the Modal (when the `closeSearch` Action is defined elsewhere). A solution would be to name Actions things like `escape` or `ctrl b`, but then there has to be a check for if the Action exists before adding to its callbacks.

Because the Modal's Action is registered second, in this example, pressing Escape would only update the SearchBar's state. The linked PR adds an option `reverseActions`, which directs the EventProcess to iterate over a KeyCombo's Actions in reverse order. This makes it easier to manage shortcuts in a temporary context which may be bound after the base site/app has initialized its shortcuts.
